### PR TITLE
WASM_X64: String hack fix

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
This PR improves the printing of string in the `wasm_x64` backends.

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
This is currently a work in progress (WIP).

**Comment by Shaikh-Ubaid** [comment]
**Example:** (It tests printing strings between integers and floating-points)
```bash
(lp) lpython$ lpython integration_tests/comp_01.py --backend wasm_x64 -o tmp > main.asm
(lp) lpython$ ./tmp
4
5
x lt y
x le y
x ne y
5
4
x gt y
x ge y
x ne y
4
6
x lt y
x le y
x ne y
5
5
x eq y
x ge y
x le y
5
5
x eq y
x ge y
x le y
4.55499999
15.55000000
x lt y
x le y
x ne y
15.55000000
4.55499999
x gt y
x ge y
x ne y
4.55499999
25.12000000
x lt y
x le y
x ne y
15.55000000
15.55000000
x eq y
x ge y
x le y
15.55000000
15.55000000
x eq y
x ge y
x le y
-4
-5
x gt y
x ge y
x ne y
-5
-4
x lt y
x le y
x ne y
-4
-6
x gt y
x ge y
x ne y
-5
-5
x eq y
x ge y
x le y
-5
-5
x eq y
x ge y
x le y
-4.55499999
-15.55000000
x gt y
x ge y
x ne y
-15.55000000
-4.55499999
x lt y
x le y
x ne y
-4.55499999
-25.12000000
x gt y
x ge y
x ne y
-15.55000000
-15.55000000
x eq y
x ge y
x le y
-15.55000000
-15.55000000
x eq y
x ge y
x le y
(lp) lpython$ 
(wasm_asm) lpython$ nasm -fbin main.asm && chmod +x main && ./main
4
5
x lt y
x le y
x ne y
5
4
x gt y
x ge y
x ne y
4
6
x lt y
x le y
x ne y
5
5
x eq y
x ge y
x le y
5
5
x eq y
x ge y
x le y
4.55499999
15.55000000
x lt y
x le y
x ne y
15.55000000
4.55499999
x gt y
x ge y
x ne y
4.55499999
25.12000000
x lt y
x le y
x ne y
15.55000000
15.55000000
x eq y
x ge y
x le y
15.55000000
15.55000000
x eq y
x ge y
x le y
-4
-5
x gt y
x ge y
x ne y
-5
-4
x lt y
x le y
x ne y
-4
-6
x gt y
x ge y
x ne y
-5
-5
x eq y
x ge y
x le y
-5
-5
x eq y
x ge y
x le y
-4.55499999
-15.55000000
x gt y
x ge y
x ne y
-15.55000000
-4.55499999
x lt y
x le y
x ne y
-4.55499999
-25.12000000
x gt y
x ge y
x ne y
-15.55000000
-15.55000000
x eq y
x ge y
x le y
-15.55000000
-15.55000000
x eq y
x ge y
x le y
(wasm_asm) lpython$ 
```

**Comment by Shaikh-Ubaid** [comment]
**Another Example:** (with slightly longer strings which contain special characters)
```bash
(lp) lpython$ cat examples/expr2.py 
def main0():
    x: i32
    x = (2+3)* (-5)
    y: i32 = 100
    z: i32 = 2147483647
    w: i32 = -2147483648
    print(x)
    print(y)
    print("Hakljfdlksjfu14578-90@!#%56&*(&_Q#<>iHelloHow are you!12343289jklsdfjas!@jdlkfaj")
    print(z)
    print(w)
    print("HiHelloHow are you!12343289jklsdfjas!@jdlkfaj")
    print("Hello")
    print((5-2) * 7)
    print("Bye")

main0()
(lp) lpython$ lpython examples/expr2.py --backend wasm_x64 -o tmp > main.asm
(lp) lpython$ ./tmp
-25
100
Hakljfdlksjfu14578-90@!#%56&*(&_Q#<>iHelloHow are you!12343289jklsdfjas!@jdlkfaj
2147483647
-2147483648
HiHelloHow are you!12343289jklsdfjas!@jdlkfaj
Hello
21
Bye
(lp) lpython$ 
(wasm_asm) lpython$ nasm -fbin main.asm && chmod +x main && ./main
-25
100
Hakljfdlksjfu14578-90@!#%56&*(&_Q#<>iHelloHow are you!12343289jklsdfjas!@jdlkfaj
2147483647
-2147483648
HiHelloHow are you!12343289jklsdfjas!@jdlkfaj
Hello
21
Bye
(wasm_asm) lpython$ 
```

File: src/libasr/codegen/wasm_to_x64.cpp [hunk 1]
```diff
@@ -430,15 +414,12 @@ class X64Visitor : public WASMDecoder<X64Visitor>,
...
+        std::string master_string = "";
         for (uint32_t i = 0; i < data_segments.size(); i++) {
-            offset = data_segments[i].insts_start_index;
-            decode_instructions();
-            std::string label = "string" + std::to_string(last_vis_i32_const);
-            label_to_str[label] = data_segments[i].text;
+            master_string += data_segments[i].text;
         }
-        decoding_data_segment = false;
+        label_to_str["master_string"] = master_string;
```
Comment (Shaikh-Ubaid) [inline src/libasr/codegen/wasm_to_x64.cpp:422]: **Note:** This approach works as we do not yet have support for arrays in the `wasm_x64` backend. When we start supporting compile-time arrays in assembly, we might need to change/update the `wasm` backend or `wasm_x64` backend.

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please possibly review and share feedback.

**Review by certik — APPROVED** [review]
I think this is an improvement, thanks!

**Pull request merged**